src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -303,6 +304,15 @@ export class NotebookModel extends Disposable implements INotebookModel {
 			if (contents) {
 				this._defaultLanguageInfo = contents.metadata && contents.metadata.language_info;
 				this._savedKernelInfo = this.getSavedKernelInfo(contents);
+				if (contents.metadata) {
+					Object.keys(contents.metadata).forEach(key => {
+						let expectedKeys = ['kernelspec', 'language_info', 'tags'];
```

Comment:
Suggest putting this in a const by the interface so that if a property is added it's less likely people will forget to update this